### ISSUE TITLE
MongoError: Index

### ISSUE THREAD

**Issue by yostools** [typeorm/typeorm#4267]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[x] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[x] `0.2.17` (or put your version here)

**MongoDB version: 4.0.3_1**

**Steps to reproduce or a small repository showing the problem:**

I use TypeORM together with NestJS (6.3.1) and MongoDB (4.0.3). Since I updated to the latest version of TypeORM (0.2.18), the server can't be started anymore.

I get the following error message with an existing database:

```
MongoError: Index with name: IDX_e12875dfb3b1d92d7d7c5377e2 already exists with different options
```

As soon as I delete the database and try to restart the server, I get the following error message:

```
MongoError: Index keys cannot be empty.
```

In version 0.2.16 of TypeORM everything still works. As soon as I use 0.2.17 or 0.2.18 the errors occur.

I can avoid the error by commenting out the block from line 51 to 57 in `src/schema-builder/MongoSchemaBuilder.ts`:

```
 metadata.uniques.forEach(unique => {
                const options = <MongodbIndexOptions>{
                    name: unique.name,
                    unique: true,
                };
                promises.push(queryRunner.createCollectionIndex(metadata.tableName, unique.columnNamesWithOrderingMap, options));
});
```

The block was added in 0.2.17 to fix #3624, but now leads to the described errors.

**Comment by vlapo** [comment]
@rustamwin could you please check and test this issue?  Thnx

First error looks like side effect of #3624 but second one is really weird. We do not have a test for #3624 so it is possible our fault.

**Comment by rustamwin** [comment]
Hey @yostools could you provide small repository to reproduce the error.

**Comment by yostools** [comment]
Please excuse the long reaction time, but I had to take a break because of an illness. The following repository uses `typeorm` version 0.2.16: https://github.com/lenneTech/nest-server 

The server can be started via `npm start` (if a MongoDB is available on localhost on port 27017) and tested via `npm run test:e2e`.

As soon as `typeorm` is included in version 0.2.17 or higher, the above mentioned bugs occur.

**Comment by yostools** [comment]
@rustamwin have you had a chance to test the repository? Can I support you in solving the problem in any way?

**Comment by rustamwin** [comment]
@yostools yes I did test your repo.
Looks like you have relation but mongo relations doesn't supported right now.
Without relations works as well.

**Comment by yostools** [comment]
@rustamwin many thanks for the hint. Is it planned to support relations for MongoDB soon?

**Comment by rustamwin** [comment]
Yes. I planned in #3472

**Comment by elie222** [comment]
Seeing the same issue, but we don't make use of relations. Error message:

> Index with name: IDX_eee922a0180dbd82621832fa08 already exists with different options

**Comment by rustamwin** [comment]
@elie222 Is this error occur in sync mode? with an existing database

**Comment by elie222** [comment]
Yes, I don't actually know what this setting does though. This is `ormconfig.json` file:

```json
{
  "type": "mongodb",
  "database": "qprivacy",
  "synchronize": true,
  "logging": false,
  "entities": ["src/**/*Entity.ts"],
  "migrations": ["src/migration/**/*.ts"],
  "subscribers": ["src/subscriber/**/*.ts"],
  "cli": {
    "entitiesDir": "src/entity",
    "migrationsDir": "src/migration",
    "subscribersDir": "src/subscriber"
  }
}
```

**Comment by elie222** [comment]
Ah I see:

> synchronize - Indicates if database schema should be auto created on every application launch. Be careful with this option and don't use this in production - otherwise you can lose production data. This option is useful during debug and development. As an alternative to it, you can use CLI and run schema:sync command. Note that for MongoDB database it does not create schema, because MongoDB is schemaless. Instead, it syncs just by creating indices.

So if I turn this, how do I resync after a schema update?

**Comment by rustamwin** [comment]
> So if I turn this off, how do I resync after a schema update?

schema:sync does this
